server/src/main/java/org/elasticsearch/cluster/routing/allocation/Explanations.java

```diff
@@ -61,6 +65,11 @@ public static final class Rebalance {
         public static final String YES = """
             Elasticsearch can rebalance this shard to another node.""";
 
+        public static final String NOT_PREFERRED = """
+            Elasticsearch will not rebalance this shard to another node because all other eligible nodes have high resource usage. The \
+            total cluster balance weights might improve, were the shard relocated, but it would push one resource usage dimension \
+            too high and threaten performance. See the node-by-node explanation to understand what resource usage is high.""";
```

Comment:
This and the above description sounds over-fitted to the write-load decider. The index balance decider will also return `NOT_PREFERRED` right? and that has nothing to do with resources?

Reply:
The index balance decider is also interested in resources, in the sense that a spike in an index' ingest would be more safely distributed across as many nodes as possible. Those are the lines along which I was thinking.

I've tweaked the phrasing a little, but not sure if as much as desirable. If you have any suggestions, let me know.